weave/trace_server/migrations/002_feedback.up.sql

```diff
@@ -0,0 +1,49 @@
+
+CREATE TABLE feedback (
+    /*
+    `id`: The unique identifier for the feedback. This is a UUID.
+    */
+    id String,
+
+    /*
+    `project_id`: The project identifier for the call. This is an internal
+    identifier that matches the project identifier in the W&B API.
+    It is stored for feedback to allow efficient permissions filtering.
+    */
+    project_id String,
+
+    /*
+    `call_id`: The unique identifier for the call. This is typically a UUID.
+    */
+    call_id String,
+
+    /*
+    `wb_user_id`: The ID of the user that created the call. This is the ID of the user in the
+    W&B API.
+    */
+    wb_user_id String NULL,
+
+    /*
+    `created_at`: The time that the row was inserted into the database.
+    */
+    created_at DateTime64(3) DEFAULT now64(3),
+
+    /*
+    `feedback_type`: The type of feedback that was given. The prefix "wandb." is reserved for our use.
+    */
+    feedback_type String,
+
+    /*
+    `notes`: An optional comment associated with the feedback.
+    */
+    notes String NULL,
+
+    /*
+    `feedback`: A dictionary of values that represent the feedback.
+    The schema of this dictionary is determined by the feedback_type.
+    */
+    feedback String NULL,
+
+
+) ENGINE = ReplacingMergeTree()
+ORDER BY (project_id, call_id, created_at);
```

Comment:
`ORDER BY` in clickhouse has a very specific meaning. It is actually very close to the meaning of "primary key": https://clickhouse.com/docs/en/engines/table-engines/mergetree-family/replacingmergetree ... we should not put `created_at` here, despite what intuition would tell us. 

Instead, you want to use a compound key of monotnically increasing cardinality. Refer to https://clickhouse.com/docs/en/guides/creating-tables#a-brief-intro-to-primary-keys and https://clickhouse.com/docs/en/optimize/sparse-primary-indexes#generic-exclusion-search-algorithm ... specifically the section starting with: `The following is illustrating how the [ClickHouse generic exclusion search algorithm](https://github.com/ClickHouse/ClickHouse/blob/22.3/src/Storages/MergeTree/MergeTreeDataSelectExecutor.cpp#L1438) works when granules are selected via a secondary column where the predecessor key column has a low(er) or high(er) cardinality.`

All this to be said, i recommend the ORDER BY to be `(project_id, call_id, wb_user_id, id)`